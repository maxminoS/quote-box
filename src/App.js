import React, { Component } from "react";
import 'App.css';
import QuoteBox from "components/QuoteBox";

class App extends Component {
  render() {
    return (
      <div className="App">
        <QuoteBox/>
      </div>
    );
  }
}

export default App;
