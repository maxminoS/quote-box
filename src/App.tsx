import React, { useState } from "react";
import { Grid, Paper, IconButton } from "@material-ui/core";
import { ThemeProvider, createMuiTheme } from "@material-ui/core/styles";
import { Brightness3, WbSunny } from "@material-ui/icons";
import 'fontsource-roboto';

import QuoteBox from "components/QuoteBox";

export default () => {
  const [darkMode, setDarkMode] = useState(false);
  const theme = createMuiTheme({ palette: { type: darkMode ? "dark" : "light" } });

  return (
    <ThemeProvider theme={theme}>
    <Paper>
      <Grid container
            justify="center"
            alignItems="center"
            style={{ minHeight: "100vh" }}
      >
        <Grid item xs={8} lg={6} xl={4}>
          <QuoteBox />
        </Grid>
        <Grid item
              style={{
                position: "absolute",
                bottom: theme.spacing(2),
                right: theme.spacing(2)
              }}
        >
          <IconButton onClick={() => setDarkMode(!darkMode)}>
            { darkMode ? <Brightness3 /> : <WbSunny /> }
          </IconButton>
        </Grid>
      </Grid>
    </Paper>
    </ThemeProvider>
  );
}
