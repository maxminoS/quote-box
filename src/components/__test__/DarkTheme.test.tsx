import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { App }from "App";

it('renders QuoteBox in dark theme as expected', () => {
  const { getByTestId, asFragment } = render(<App />);
  userEvent.click(getByTestId('theme-button'));
  expect(asFragment()).toMatchSnapshot();
});
