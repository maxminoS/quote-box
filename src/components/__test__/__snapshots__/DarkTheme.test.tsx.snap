// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders QuoteBox in dark theme as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
      style="min-height: 100vh;"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8 MuiGrid-grid-lg-6 MuiGrid-grid-xl-4"
      >
        <div
          class="MuiPaper-root MuiCard-root MuiPaper-elevation8 MuiPaper-rounded"
        >
          <div
            class="MuiCardContent-root"
          />
          <div
            class="MuiCardActions-root MuiCardActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary"
              style="margin-left: auto;"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                GET
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item"
        style="position: absolute; bottom: 8px; right: 8px;"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="theme-button"
          tabindex="0"
          title="Dark Mode"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M9 2c-1.05 0-2.05.16-3 .46 4.06 1.27 7 5.06 7 9.54 0 4.48-2.94 8.27-7 9.54.95.3 1.95.46 3 .46 5.52 0 10-4.48 10-10S14.52 2 9 2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          >
            <span
              class="MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
              style="width: 1px; height: 1px; top: -0.5px; left: -0.5px;"
            >
              <span
                class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
              />
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
